rllib/algorithms/algorithm.py

```diff
@@ -3188,633 +3189,703 @@ def validate_env(env: EnvType, env_context: EnvContext) -> None:
+            # TODO: Remove this key at some point. Here for backward compatibility.
+            eval_results["timesteps_this_iter"] = eval_results.get(
+                NUM_ENV_STEPS_SAMPLED_THIS_ITER, 0
+            )
+        else:
+            self.metrics.merge_and_log_n_dicts(
+                all_metrics,
+                key=(EVALUATION_RESULTS, ENV_RUNNER_RESULTS),
+            )
+            eval_results = self.metrics.reduce((EVALUATION_RESULTS, ENV_RUNNER_RESULTS))
 
-        # Old API stack: Policies are in separate sub-dirs.
-        if (
-            checkpoint_info["checkpoint_version"] > version.Version("0.1")
-            and state.get("worker") is not None
-            and state.get("worker")
-        ):
-            worker_state = state["worker"]
+        # Warn if results are empty, it could be that this is because the eval timesteps
+        # are not enough to run through one full episode.
+        if eval_results[ENV_RUNNER_RESULTS][NUM_EPISODES] == 0:
+            logger.warning(
+                "This evaluation iteration resulted in an empty set of episode summary "
+                "results! It's possible that your configured duration timesteps are not"
+                " enough to finish even a single episode. Your have configured "
+                f"{self.config.evaluation_duration}"
+                f"{self.config.evaluation_duration_unit}. For 'timesteps', try "
+                "increasing this value via the `config.evaluation(evaluation_duration="
+                "...)` OR change the unit to 'episodes' via `config.evaluation("
+                "evaluation_duration_unit='episodes')` OR try increasing the timeout "
+                "threshold via `config.evaluation(evaluation_sample_timeout_s=...)` OR "
+                "you can also set `config.evaluation_force_reset_envs_before_iteration`"
+                " to False. However, keep in mind that in the latter case, the "
+                "evaluation results may contain some episode stats generated with "
+                "earlier weights versions."
+            )
 
-            # Retrieve the set of all required policy IDs.
-            policy_ids = set(
-                policy_ids if policy_ids is not None else worker_state["policy_ids"]
+        # After evaluation, do a round of health check on remote eval workers to see if
+        # any of the failed workers are back.
+        # Add number of healthy evaluation workers after this iteration.
+        eval_results[
+            "num_healthy_workers"
+        ] = self.eval_env_runner_group.num_healthy_remote_workers()
+        eval_results[
+            "num_in_flight_async_reqs"
+        ] = self.eval_env_runner_group.num_in_flight_async_reqs()
+        eval_results[
+            "num_remote_worker_restarts"
+        ] = self.eval_env_runner_group.num_remote_worker_restarts()
+
+        return train_results, {"evaluation": eval_results}, train_iter_ctx
+
+    def _run_offline_evaluation(self):
```

Comment:
Oh, we have to implement this one for the new stack

Reply:
yup, I guess. If we want to support these evaluation algos. I'm not sure yet how much of a priority this should be, though ...